File: 차이를최대로.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;
int calDiffSum(vector<int>&a){
  int ret= 0; 
  for(int i=1; i<a.size(); i++){
    ret +=abs(a[i]-a[i-1]);
  }
  return ret;
}
int main(){
  int n ;
  cin>>n;
  vector<int> arr(n);
  for(int i=0; i<n; i++){
    cin>>arr[i];
  }
  sort(arr.begin(), arr.end());
  int ans = 0;
  do{
    ans =  max(ans,calDiffSum(arr));
  }while(next_permutation(arr.begin(), arr.end()));
  cout<<ans<<'\n';
}

File: 차이를최대로.py
from itertools import permutations
def calDiffSum(a):
  ret = 0
  for i in range(1,len(a)):
    ret+= abs(a[i]-a[i-1])
  return ret
n=int(input())
arr = list(map(int,input().split()))
arr.sort()
ans =0
for perm in list(permutations(arr)):
  ans = max(ans,calDiffSum(perm))
print(ans)